crates/rbuilder/src/live_builder/mod.rs

```diff
@@ -215,12 +229,49 @@ impl<DB: Database + Clone + 'static, BuilderSourceType: SlotSource>
                 None,
             );
 
-            builder_pool.start_block_building(
-                payload,
-                block_ctx,
-                self.global_cancellation.clone(),
-                time_until_slot_end.try_into().unwrap_or_default(),
-            );
+            // This was done before in block building pool
+            {
+                let max_time_to_build = time_until_slot_end.try_into().unwrap_or_default();
+                let block_cancellation = self.global_cancellation.clone().child_token();
+
+                let cancel = block_cancellation.clone();
+                tokio::spawn(async move {
+                    tokio::time::sleep(max_time_to_build).await;
+                    cancel.cancel();
+                });
+
+                let (orders_for_block, sink) = OrdersForBlock::new_with_sink();
+                // add OrderReplacementManager to manage replacements and cancellations
+                let order_replacement_manager = OrderReplacementManager::new(Box::new(sink));
+                // sink removal is automatic via OrderSink::is_alive false
+                let _block_sub = orderpool_subscriber.add_sink(
+                    block_ctx.block_env.number.to(),
+                    Box::new(order_replacement_manager),
+                );
+
+                let simulations_for_block = order_simulation_pool.spawn_simulation_job(
+                    block_ctx.clone(),
+                    orders_for_block,
+                    block_cancellation.clone(),
+                );
+
+                let (broadcast_input, _) = broadcast::channel(10_000);
+                let builder_sink = sink_factory.create_sink(payload, block_cancellation.clone());
+
+                let input = BlockBuildingAlgorithmInput::<DB> {
+                    provider_factory: self.provider_factory.provider_factory_unchecked(),
+                    ctx: block_ctx,
+                    sink: builder_sink,
+                    input: broadcast_input.subscribe(),
+                    cancel: block_cancellation,
+                };
+
+                tokio::spawn(multiplex_job(simulations_for_block.orders, broadcast_input));
+
+                if let Some(builder) = self.builder.as_ref() {
+                    builder.build_blocks(input);
```

Comment:
Even though you know that BlockBuildingPool is not blocking, builders are expected to be blocking. We should tokio::task::spawn_blocking this.

Reply:
Which one do you want as spawn_blocking?

Reply:
builder.build_blocks(input);

Reply:
Check again, I think this is resolved now.